src/stores/HistoryStore.js

```diff
@@ -70,20 +76,138 @@ export default class {
     reaction(
       () => this.app.global.syncBlockNum,
       async () => {
-        if (this.transactions.length > 0 && this.app.ui.location === Routes.ACTIVITY_HISTORY) {
-          const { naka: { account } } = this.app;
-          const filters = { transactorAddress: account };
-          const newTxs = await this.fetchHistory(filters, ACTIVITY_HISTORY_LIMIT, 0, 0, 0, 0, 0, true);
-          const old = this.transactions[0];
-          if (newTxs.length > 0 && (newTxs[0].txid !== old.txid
-            || newTxs[0].txStatus !== old.txStatus)) {
-            this.transactions.splice(0, 1, newTxs[0]);
-          }
-        }
+        if (!this.updating) await this.update();
       },
     );
   }
 
+  // update skips so when loading more so that duplicate txs won't appear
+  updateSkips = (tx, type) => {
```

Comment:
Right...it's bug prone and hard to maintain.

I've been thinking about changing all of these. Instead of passing skips to backend and backend returns skips, maybe using timestamp to do the transactions query is much easier?

Reply:
But this version does solve issues, I think it can be merged first, and optimize later.